### PR TITLE
Shared-state in soporteremoto

### PR DESCRIPTION
**Pull request by gmarcos87**
WIP

### CODE REVIEW

File: federated-server/shared-state.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export async function getActualNodes(){
+    //Merge node data into my shared-state
+    const nodeData = await requestSharedState('babeld-hosts', false);
+    myState.merge(objToStrMap(nodeData) ,false,'babeld-hosts');
+    //Format result to get a hostname list
+    let nodes = []
+    myState
+        .show('babeld-hosts')
+        .forEach(babledHost => nodes = nodes.indexOf(babledHost.data) !== -1 ? nodes: [...nodes, babledHost.data])
```
Comment (germanferrero) [inline federated-server/shared-state.js]: There is a typo here: babledHost. And also I just checked that we have [Sets](http://javascript.info/map-set#set) in JS, if you want to use it as a shortcut.

File: federated-server/shared-state.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
+const SharedState = require("shared-state-js");
+const { strMapToObj, objToStrMap } = require('obj2map');
```
Comment (germanferrero) [inline federated-server/shared-state.js]: I'm curious about these two guys, isn't there native support for this functionality?

File: federated-server/shared-state.js (hunk 1 above)
Comment (gmarcos87) [reply federated-server/shared-state.js]: done

File: federated-server/shared-state.js (hunk 2 above)
Comment (gmarcos87) [reply federated-server/shared-state.js]: When I tried at the time I didn't find a better way or a native way. That is a library of my authorship that implements the two steps in 15 lines of very basic code.

**Pull request merged**